include/llarp/dns_rectypes.hpp

```diff
@@ -0,0 +1,114 @@
+#ifndef LIBLLARP_DNS_REC_TYPES_HPP
+#define LIBLLARP_DNS_REC_TYPES_HPP
+
+#include <vector>
+#include <llarp/buffer.h>  // for byte_t
+#include <llarp/net.hpp>   // for llarp::Addr , llarp::huint32_t
+
+namespace llarp
+{
+  namespace dns
+  {
+    struct record
+    {
+      virtual ~record(){};
```

Comment:
If this class shouldn't be instantiated, tag the destructor as `virtual ~record() = 0;`, and define out-of-line.

Reply:
ah yea, tried copying the inheritance pattern from bencode.hpp, message.hpp and findintro.hpp but because of the 3 layers was a bit harder to follow.

Changed the destructor. Not sure what you mean by out-of-line.